cmake/CMakeLists.txt

```
@@ -612,6 +614,18 @@ else()
      message(FATAL_ERROR  "The compiler doesn't support FLOAT16!!!")
    endif()
   endif()
+
+  if(onnxruntime_USE_SVE)
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64" AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
+      check_cxx_compiler_flag("-march=armv8.2-a+sve" HAS_ARM64_SVE)
+      if(HAS_ARM64_SVE)
+        message(STATUS "Compiler supports SVE!")
+      else()
+        message(STATUS "Compiler does not support SVE!")
+      endif()
+    endif()
+  endif()
```

how about updating `onnxruntime_USE_SVE` to be `OFF` if it is not supported? then we can simply check for `onnxruntime_USE_SVE` later on.
```suggestion
  if(onnxruntime_USE_SVE)
    if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64" AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
      check_cxx_compiler_flag("-march=armv8.2-a+sve" HAS_ARM64_SVE)
      if(HAS_ARM64_SVE)
        message(STATUS "Compiler supports SVE!")
      else()
        message(WARNING "onnxruntime_USE_SVE was set but compiler does not support SVE. It will be disabled.")
        set(onnxruntime_USE_SVE OFF)
      endif()
    else()
      message(WARNING "onnxruntime_USE_SVE was set but it is not supported on this platform. It will be disabled.")
      set(onnxruntime_USE_SVE OFF)
    endif()
  endif()
```

---

also, I noticed that this code is within an `else()`. can it be moved outside so that it always runs?

```
if(WIN32)
  ...
else()
  <this check>
endif()
```